### ISSUE TITLE
Should the path be normalized?

### ISSUE THREAD

**Issue by DLMR** [koajs/send#51]
I'm using `koa-send` in my Koa server and I have the problem that I get `Malicious Path` from `resolve-path` when the request path is something along the lines of `http://localhost:3000//some/path/image.jpg`.

Would it be a bad idea to normalize the path in `koa-send` before processing it further? This could for instance be done here https://github.com/koajs/send/blob/master/index.js#L46 before doing `substring` on the path.

**Comment by Pumpuli** [comment]
I'm on Windows and I get the same error with all requests because `resolve-path` thinks they are absolute paths. That is because the path always starts with a slash. I assume [this line](https://github.com/koajs/send/blob/58bf426ddd366818a8261df6f5ed17773ad3860e/index.js#L46) is supposed to remove the leading slash, but on Windows `path.parse` returns `''` for root for paths that start with a slash. Therefore the path will have a leading slash when it's passed to `resolve-path`.

**Comment by coderhaoxin** [comment]
> but on Windows path.parse returns '' for root for paths that start with a slash

test with `node@4` on win10(VM)

```js
path.parse('/a/b')
```

```js
{root: '/', ...}
```

**Comment by coderhaoxin** [comment]
@Pumpuli I think your problem is https://github.com/koajs/static/issues/77 ?

**Comment by Pumpuli** [comment]
> I think your problem is koajs/static#77 ?

Yep, seems to be a Node issue, works on Node 5.6.0.

**Comment by zbinlin** [comment]
@coderhaoxin
In Node 5.7.0

    path.parse("/a/b")

returns

    { root: '', dir: '/a', base: 'b', ext: '', name: 'b' }

I suggest use `path.posix.parse` instead of `path.parse`.

**Comment by coderhaoxin** [comment]
@zbinlin I saw your PR to node, `5.7.1` will fix this. right?

**Comment by zbinlin** [comment]
@coderhaoxin right, but I also suggest that use `path.posix.parse` instead of `path.parse` in koa-send.

**Comment by tejasmanohar** [comment]
>I also suggest that use `path.posix.parse` instead of `path.parse` in koa-send.
@zbinlin why so? shouldn't we use _path#parse_ and let Node determine how to resolve the path dependent on the environment.

**Comment by zbinlin** [comment]
@tejasmanohar I think the path is usually unix path(or url path), so we can only remove first forward slash of the path use `path.posix.parse`;

**Comment by Pumpuli** [comment]
Yeah, I think the point is that `path#parse` is for parsing filesystem paths, but in this case we're parsing the path part of a URL.
